src/Promenade.Controllers/HomeController.cs

```diff
@@ -1,21 +1,362 @@
+                    {
+                        try
+                        {
+                            var apikey = _config["APIKey"];
+
+                            var response = await client.GetAsync($"https://maps.googleapis.com/maps/api/geocode/json?address={address}&key={apikey}");
+                            response.EnsureSuccessStatusCode();
+
+                            var stringResult = await response.Content.ReadAsStringAsync();
+                            dynamic jsonResult = JsonConvert.DeserializeObject(stringResult);
+
+                            if (jsonResult.results.Count > 0)
+                            {
+                                var result = jsonResult.results[0];
+                                latitude = result.geometry.location.lat;
+                                longitude = result.geometry.location.lng;
+
+                                Location viewModel = LibraryLookup(latitude, longitude);
+                                viewModel.FormattedAddress = result.formatted_address;
+                                if (mobile == 1)
+                                {
+                                    viewModel.ShowLocation = true;
+                                }
+                                return View("Locations", viewModel);
+                            }
+                            else
+                            {
+                                _logger.LogError($"No geocoding results for address \"{address}\"");
+                                TempData["AlertDanger"] = $"Unable to locate address <strong>\"{address}\"</strong>.";
+                            }
+                        }
+                        catch (HttpRequestException ex)
+                        {
+                            _logger.LogCritical(ex, $"Google API error: {ex.Message}");
+                            TempData["AlertDanger"] = "An error occured, please try again later.";
+                        }
+                        catch (Exception ex)
+                        {
+                            _logger.LogCritical(ex, ex.Message);
+                            TempData["AlertDanger"] = "An error occured, please try again later.";
+                        }
+                    }
+                }
+                return View("Locations");
+            }
+            else
+            {
+                var viewModel = new Location();
+                viewModel.CloseLocations = locations.OrderBy( c => c.Name).ToList();
+                if (mobile==1)
+                {
+                    viewModel.ShowLocation = true;
+                }
+                return View("Locations", viewModel);
+            }
+        }
+
+        [HttpGet("[action]/{locationStub}")]
+        [HttpGet("[action]/{locationStub}/{featureStub}")]
+        public IActionResult Locations(string locationStub, string featureStub)
+        {
+            Location[] locations = {  };
+
```

Comment:
No reason to have two lines of whitespace here, only ever one blank line between lines of code.